### PR TITLE
mpd 0.22

### PR DESCRIPTION
**Pull request by 00dani**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by 00dani** [comment]
Ugh. CI fails because as of mpd 0.22, building now depends on C++17. It built fine locally, presumably because I'm running on an 11.0 beta, but I don't know how to change the formula so that it requests C++17 support. 😕

**Comment by SMillerDev** [comment]
I think that would be changing `ENV.libcxx` to `ENV.cxx17` but I'm not sure.

**Comment by 00dani** [comment]
Unfortunately it looks like that doesn't work:
```
Error: An exception occurred within a child process:
  NoMethodError: undefined method `cxx17' for #<Object:0x00007fed1f8946f8>
Did you mean?  cxx11
               cxx
```
Hmm.

**Comment by SMillerDev** [comment]
@Homebrew/maintainers idens?

**Comment by EricFromCanada** [comment]
`-Dcpp_std=c++17` should do that for Meson, but in the existing 10.13 test output, it looks like `clang++` is already using the `-std=c++17` flag.

**Comment by 00dani** [comment]
Hmm, you're right, I can see the `-std=c++17` flag in the output. Perhaps not using C++17 _isn't_  the problem, then?

I don't know how else to interpret this error from clang++, though:
```
../../src/ls.cxx:46:14: error: no member named 'merge' in 'std::__1::set<std::__1::basic_string<char>, std::__1::less<std::__1::basic_string<char> >, std::__1::allocator<std::__1::basic_string<char> > >'
                        protocols.merge(plugin.protocols());
                        ~~~~~~~~~ ^
```
It definitely seems that `std::set<K,C,A>::merge` was [added in C++17](https://en.cppreference.com/w/cpp/container/set/merge), which suggests its absence means C++17 isn't being used. 😕

**Comment by jonchang** [comment]
macOS clang doesn't always implement all of the features in the standard. It's possible that this is one of those cases, but I haven't investigated this specific case in detail.

File: Formula/mpd.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Mpd < Formula
   desc "Music Player Daemon"
   homepage "https://www.musicpd.org/"
-  url "https://www.musicpd.org/download/mpd/0.21/mpd-0.21.25.tar.xz"
-  sha256 "20a0ad01bf327b2dbeb6ae8e1af952cb0de83d2d63fab0fa4b7183a74765c201"
+  url "https://www.musicpd.org/download/mpd/0.22/mpd-0.22.tar.xz"
+  sha256 "7de742d925d01bf52685cc6cbb31fdeb671e4992a448720897ef706e57e39076"
   license "GPL-2.0"
```
Comment (equal-l2) [inline Formula/mpd.rb:6]: According to its license header (in https://github.com/MusicPlayerDaemon/MPD/blob/master/src/BulkEdit.hxx, for example):
```suggestion
  license "GPL-2.0-or-later"
```

**Comment by 00dani** [comment]
It looks like the relevant C++17 feature is called "Splicing Maps and Sets", and [apparently it's available as of Apple Clang 10.0.0](https://en.cppreference.com/w/cpp/compiler_support#C.2B.2B17_library_features). All three CI environments have at least Clang 10.0.0, so I'm very confused. 🤔

**Comment by 00dani** [comment]
Curiously, after I made that second commit that simply changes the formula's license information, it built successfully on 10.14 and 10.15. No changes to the actual build configuration. So that's weird?

Still no luck with 10.13 though. 😕

**Comment by jonchang** [comment]
We updated to Xcode 12. Can you add a dependency to Mojave and above? It should skip building on High Sierra once you do that.

**Comment by fxcoudert** [comment]
Thanks @00dani, merging now

**Pull request closed without merging**